assets/src/js/Components/archivePicker/SingleSidebar.js

```diff
@@ -1,135 +1,114 @@
-import { Component, Fragment } from '@wordpress/element';
-import { getSizesUnder, toSrcSet } from '../ImagePicker';
+import { useState, useEffect, Fragment } from '@wordpress/element';
+import { getSizesUnder, toSrcSet } from './sizeFunctions';
 
 const { __ } = wp.i18n;
 
 const PREVIEW_MAX_SIZE = 1300;
 
-const wpImageLink = ( id ) => `${ window.location.href.split( '/wp-admin' )[ 0 ] }/wp-admin/post.php?post=${ id }&action=edit`;
+const wpImageLink = id => `${window.location.href.split('/wp-admin')[0]}/wp-admin/post.php?post=${id}&action=edit`;
 
-const largestSize = ( image ) => !image ? null : image.original;
+const largestSize = image => !image ? null : image.original;
 
-const renderDefinition = ( key, value ) => ( <div>
-  <dt>{ key }</dt>
-  <dd>{ value }</dd>
-</div> );
+const renderDefinition = (key, value) => (
+  <div>
+    <dt>{key}</dt>
+    <dd>{value}</dd>
+  </div>
+);
 
-export class SingleSidebar extends Component {
-  constructor( props ) {
-    super( props );
-    this.state = {
-      imageLoaded: false,
-    };
-    this.renderImage = this.renderImage.bind(this);
-    this.preloadImage = this.preloadImage.bind(this);
-  }
+export const SingleSidebar = ({ image, processingError, processingImages, includeInWp }) => {
+  const [imageLoaded, setImageLoaded] = useState(false);
 
-  componentDidMount() {
-    this.preloadImage();
-  }
+  const original = largestSize(image) || {};
+  const aspectRatio = original.height / original.width;
 
-  async componentDidUpdate( prevProps ) {
-    if ( prevProps.image !== this.props.image ) {
-      this.preloadImage();
+  const preloadImage = async () => {
+    if (image) {
+      setImageLoaded(false);
+      await fetch(largestSize(getSizesUnder(image.sizes, PREVIEW_MAX_SIZE)));
```

Comment:
`getSizesUnder()` returns an array of `image` objects, which does not seem to fit `largestSize()`; fetch here doesn't fetch anything ? I noticed this because there's an ajax call to `/undefined` on every click to select an image

Reply:
The reason I had added this was because otherwise the sidebar would render with the new data but still the old image. Then only when the image is fully loaded it would replace the previous image. This can take a long time if it's a big image.

I had first added the preloading, then later also added `getSizesUnder`, because the original image was extremely large in some cases. But looks like I broke it here as `largestSize` expects the image itself.

I didn't notice because even with the wrong fetch it still causes the previous image to be replaced with the loading placeholder.

Reply:
You can see the original issue if you initialize `imageLoaded` as true and remove the preloading effect. Click on a few images and you'll see some of the time the previous image is displayed with the data of the new one for multiple seconds.

Reply:
But it was more a quick fix at the end of the PR, I'm sure there's a better way to solve it.

Reply:
Ah yes there is indeed https://github.com/facebook/react-native/issues/9195#issuecomment-238190537

Reply:
Checked locally and this seems to work
```
            <img
              key={image.original.url}
              srcSet={toSrcSet(image.sizes, { maxWidth: PREVIEW_MAX_SIZE })}
              title={image.title}
              alt={image.title}
            />
```

Reply:
Updated! 😄